volume/mounts/mounts.go

```diff
@@ -80,6 +95,8 @@ type MountPoint struct {
 	// SafePaths created by Setup that should be cleaned up before unmounting
 	// the volume.
 	safePaths []*safepath.SafePath
+
+	Layer RWLayer `json:"-"`
```

Comment:
This value would need to get populated on `restore` similar to  https://github.com/moby/moby/blob/master/daemon/daemon.go#L246. I don't see where it get set after a daemon restart, I would suspect that remove would panic after restart when this is nil.

Reply:
@dmcgowan you are correct....I didn't see this until after I found out that it was the cause of my issue.

I added this but I am not sure it is enough

https://github.com/moby/moby/pull/48798/files#diff-6528b1001f020d3cedd8ff77da4ed71d862d00c6f1e7c11187dc33d7144a29cfR21-R30